removing memchache mentions in comments

rebeccahhh opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
just removing all mentions of memcached in comments throughout the codebase. In some cases this is replaced with the appropriate term for redis-caching.
<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Docs Pull Request 

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 12.0.0
```

rebeccahhh committed:
removing memchache mentions in comments

remove memcached folder as it is no longer needed, also address a couple grammatical errors

rebeccahhh commented on docs/tasks.md:
```diff
@@ -263,9 +263,9 @@ This entire process enables a response of a `202 Accepted`, where instead of wai
+Any time the user changes a setting in AWX (_e.g._, in `api/v2/settings`), data will be added to or altered in a database. Since querying databases directly can be extremely time-consuming, each node in a cluster runs a local `redis-cache` server, none of which are aware of each other. They all potentially have different values contained within, but ultimately need to be consistent. So how can th ...
 
-"Handle Setting Changes" provides the solution!  This "fanout" task (_i.e._, all nodes execute it) makes it so that there is a single source of truth even within a clustered system. Whenever a database setting is accessed, and that setting's name is not present in `memcached`, it grabs the setting from the database and then populates it in the applicable node's cache.  When any database setting g ...
+"Handle Setting Changes" provides the solution!  This "fanout" task (_i.e._, all nodes execute it) makes it so that there is a single source of truth even within a clustered system. Whenever a database setting is accessed, and that setting's name is not present in `redis-cache`, it grabs the setting from the database and then populates it in the applicable node's cache.  When any database setting ...
```
@chrismeyersfsu I particularly want you to take a peek at this section and validate that I am accurate. :)

beeankha commented on awx/main/tests/conftest.py:
```diff
@@ -131,8 +131,8 @@ def delete(self, key):
     # clear Django cache at the end of every test ran
-    # NOTE: this should not be memcache, see test_cache in test_env.py
-    # this is a local test cache, so we want every test to start with empty cache
+    # NOTE: this should not be memcache (as it is depricated), nor should it be redis.
```
depricated ---> deprecated

beeankha commented on awx/main/tests/conftest.py:
```diff
@@ -131,8 +131,8 @@ def delete(self, key):
-    # NOTE: this should not be memcache, see test_cache in test_env.py
-    # this is a local test cache, so we want every test to start with empty cache
+    # NOTE: this should not be memcache (as it is depricated), nor should it be redis.
+    # This is a local test cache, so we want every test to start with empty cache
```
nitpick: "with _an_ empty cache"

beeankha reviewed (approved):
Pointed out a couple of minor things, otherwise looks great!  👍

wenottingham commented on awx/plugins/inventory/ec2.py:
```diff
@@ -1201,7 +1201,7 @@ def add_elasticache_cluster(self, cluster, region):
         # Select the best destination address
         if 'ConfigurationEndpoint' in cluster and cluster['ConfigurationEndpoint']:
-            # Memcached cluster
+            # Redis cluster
```
this (and below)  is actually pulling memcached clusters from AWS in inventory, it's fine to keep.

@AlanCoding is deleting this shortly anyway.

ryanpetrello replied:
Yep, let's leave this file in `awx/plugins` untouched in this PR, it's going away in the near future anyways (https://github.com/ansible/awx/pull/6911).

ryanpetrello commented on awx/conf/settings.py:
```diff
@@ -31,18 +31,18 @@
 # of None will be equivalent to the setting not being set (and will raise an
 # AttributeError if there is no other default defined).
 # SETTING_CACHE_NONE = '___none___'
 SETTING_CACHE_NONE = SETTING_CACHE_NOTSET
```
I'm betting that we don't actually _need_ to do this sort of thing anymore - it's possible that redis and its Python driver have support for storing `None`, `[]`, and `{}`.  That said, this could be a really hairy bit of code to tease apart, and I'm kind of scared to touch it since it's not really hurting anything keeping it this way.

tl;dr if it ain't broke, don't fix it

ryanpetrello commented on awx/conf/tests/unit/test_settings.py:
```diff
@@ -66,15 +66,6 @@ def test_unregistered_setting(settings):
 
-def test_cached_settings_unicode_is_auto_decoded(settings):
-    # https://github.com/linsomniac/python-memcached/issues/79
-    # https://github.com/linsomniac/python-memcached/blob/288c159720eebcdf667727a859ef341f1e908308/memcache.py#L961
```
God I hated this test 👍

ryanpetrello commented on awx/conf/tests/unit/test_settings.py:
```diff
@@ -254,31 +245,6 @@ def test_setting_from_db(settings, mocker):
-        default='DEFAULT',
-        encrypted=encrypted
-    )
-    # this simulates a bug in python-memcached; see https://github.com/linsomniac/python-memcached/issues/79
```
...and it's ugly cousin.

ryanpetrello reviewed (approved):
A few minor comments, looks great.

ryanpetrello commented on tools/memcached/.dir_placeholder:
```diff
@@ -1 +1 @@
-This dir must pre-exist and be owned by the user you are launching awx dev env as. If the dir does not exist before launching the awx dev environment then docker will create the dir and it will be owned by root. Since we start our awx dev environment with user: ${CURRENT_UID} the memcached container will be unable to create a socket file in a directory owned by root.
+This dir must pre-exist and be owned by the user you are launching awx dev env as. If the dir does not exist before launching the awx dev environment then docker will create the dir and it will be owned by root. Since we start our awx dev environment with user: ${CURRENT_UID} Redis will be unable to create a socket file in a directory owned by root.
```
Actually, I think we can just entirely `git rm tools/memcached` my understanding is that this directory only existed as a place to mount the unix socket file that memcached was listening on cc @chrismeyersfsu

rebeccahhh commented:
squashed to one commit because this is an appropriate time to use squashing, I should have addressed all the comments, I am pleased to :wave: goodbye to memcached soon :)
